### PR TITLE
[EPROD-1099] fix signature v

### PR DESCRIPTION
**Pull request by ufoscout**
# Issue ticket

Issue ticket link: <>
https://infinityswap.atlassian.net/browse/EPROD-1099

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [x] Every function is properly unit tested
- [x] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/eth-signer/src/ic_sign.rs [hunk 1]
```diff
@@ -166,15 +170,15 @@ impl IcSigner {
 
         let r = U256::from_be_slice(&signature_data[0..32]);
         let s = U256::from_be_slice(&signature_data[32..64]);
-        let v = recovery_id.is_y_odd() as u64 + 27;
+        let y_parity = recovery_id.is_y_odd();
```
Comment (veeso) [inline src/eth-signer/src/ic_sign.rs:173]: y_parity = true. Could be misleading. Parity is even or odd

File: src/eth-signer/src/ic_sign.rs [hunk 2]
```diff
@@ -115,20 +115,24 @@ impl IcSigner {
...
-        signature.v = match tx.chain_id() {
-            Some(chain_id) => (chain_id * 2 + 35 + (v - 27)).into(),
-            None => v.into(),
-        };
+        // let mut signature = Self
+        //     .sign_digest(*hash, pubkey, key_id, derivation_path)
+        //     .await?;
 
-        Ok(signature)
+        // let v: u64 = signature.v.0.to();
```
Comment (veeso) [inline src/eth-signer/src/ic_sign.rs:126]: can we remove the commented code?

**Review by veeso — CHANGES_REQUESTED** [review]

File: src/eth-signer/src/ic_sign.rs (hunk 2 above)
Comment (ufoscout) [reply src/eth-signer/src/ic_sign.rs:126]: done

File: src/eth-signer/src/ic_sign.rs (hunk 1 above)
Comment (ufoscout) [reply src/eth-signer/src/ic_sign.rs]: done

**Review by veeso — APPROVED** [review]

**Pull request merged**